File: my-agent-framework/shared/api_schema.ts
// API Schema Types — TypeScript types for REST API
export interface ChatMessage {
  role: "user" | "assistant" | "system" | "tool";
  content: string;
}

export interface ChatRequest {
  messages: ChatMessage[];
  session_id?: string;
  stream?: boolean;
  agent_name?: string;
  temperature?: number;
}

export interface ChatResponse {
  content: string;
  session_id: string;
  model: string;
  agent_used?: string;
}

export interface Agent {
  name: string;
  description: string;
  enabled: boolean;
  tools: string[];
}

export interface ModelInfo {
  adapter: string;
  model: string;
  available: boolean;
}

export interface HealthResponse {
  status: "ok" | "degraded" | "error";
  version: string;
}

File: my-agent-framework/shared/api_schema.py
"""
api_schema.py — Pydantic Models untuk API (shared Python side)
"""
from pydantic import BaseModel
from typing import List, Literal, Optional


class ChatMessage(BaseModel):
    role: Literal["user", "assistant", "system", "tool"]
    content: str


class ChatRequest(BaseModel):
    messages: List[ChatMessage]
    session_id: Optional[str] = None
    stream: bool = False
    agent_name: Optional[str] = None
    temperature: float = 0.7


class ChatResponse(BaseModel):
    content: str
    session_id: str
    model: str
    agent_used: Optional[str] = None


class AgentInfo(BaseModel):
    name: str
    description: str
    enabled: bool
    tools: List[str] = []


class HealthResponse(BaseModel):
    status: str
    version: str
